store/views.py

```diff
@@ -1,82 +1,263 @@
-        'books': None, # set this to the list of required books upon filtering using the GET parameters
-                       # (i.e. the book search feature will also be implemented in this view)
+        'books': books,
     }
-    get_data = request.GET
-    # START YOUR CODE HERE
-    
-    
     return render(request, template_name, context=context)
 
 @login_required
 def viewLoanedBooks(request):
     template_name = 'store/loaned_books.html'
+    # Search for Loaned Books by the user
+    books = BookCopy.objects.filter(borrower = request.user)
     context = {
-        'books': None,
+        'books': books,
     }
-    '''
-    The above key 'books' in the context dictionary should contain a list of instances of the 
-    BookCopy model. Only those book copies should be included which have been loaned by the user.
-    '''
-    # START YOUR CODE HERE
-    
-
-
     return render(request, template_name, context=context)
 
 @csrf_exempt
 @login_required
 def loanBookView(request):
-    response_data = {
-        'message': None,
-    }
     '''
     Check if an instance of the asked book is available.
     If yes, then set the message to 'success', otherwise 'failure'
     '''
-    # START YOUR CODE HERE
-    book_id = None # get the book id from post data
+    if request.method == 'POST':
+        # Get the copies of the required book if it is available for loan
+        book_id = request.POST['bid']
+        books = BookCopy.objects.filter(book_id = book_id, status = True)
+
+        # If copy is available
+        if books.count() > 0:
+            msg = 'success' 
+            # Select the first copy
+            books = books.first()
+            # Set to unavailable
+            books.status = False
+            # User is the borrower
+            books.borrower = request.user
+            # Current date is borrow date
+            books.borrow_date = datetime.date.today()
+            # Save the model object
+            books.save()
+        else:
+            # The following two are possible:
+            #   1. Either no such book is found, or
+            #   2. no such book is available for loan
+            msg = 'failure'
 
+        # Return the message
+        response_data = {
+            'message': msg,
+        }
+        return JsonResponse(response_data)
+
+    # This page is not available for user to view and is just for data exchange
+    else:
+        raise Http404
 
-    return JsonResponse(response_data)
 
-'''
-FILL IN THE BELOW VIEW BY YOURSELF.
-This view will return the issued book.
-You need to accept the book id as argument from a post request.
-You additionally need to complete the returnBook function in the loaned_books.html file
-to make this feature complete
-''' 
 @csrf_exempt
 @login_required
 def returnBookView(request):
-    pass
+    # Books are returned through POST method 
+    if request.method == 'POST': 
+        # Get the desired book copy
+        id = request.POST['bid']
```

Comment:
You are directly accessing POST data without checking if it even exists. This may lead to server crash if a user access this endpoint with invalid request data. The good behavior would have been to throw a client error (400), rather than server error (500).